PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/AzureAnAction.java

```diff
@@ -52,17 +56,26 @@ public AzureAnAction(@Nullable String text, @Nullable String description, @Nulla
         super(text, description, icon);
     }
 
-    public abstract void onActionPerformed(AnActionEvent anActionEvent);
+    /**
+     * @param anActionEvent action event
+     * @param operation operation for sending telemetry
+     * @return if the action is a synchronous action, you should return true and let us complete the operation
+     * if the action is an asynchronous action, you should return false and control the operation completion by yourself
+     */
+    public abstract boolean onActionPerformed(@NotNull AnActionEvent anActionEvent, @Nullable Operation operation);
 
     @Override
-    public final void actionPerformed(AnActionEvent anActionEvent) {
+    public final void actionPerformed(@NotNull AnActionEvent anActionEvent) {
         sendTelemetryOnAction(anActionEvent, "Execute", null);
-        String serviceName = transformHDInsight(getServiceName(), anActionEvent);
+        String serviceName = transformHDInsight(getServiceName(anActionEvent), anActionEvent);
         String operationName = getOperationName(anActionEvent);
-        EventUtil.executeWithLog(serviceName, operationName, (operation) -> {
-            EventUtil.logEvent(EventType.info, operation, buildProp(anActionEvent, null));
-            onActionPerformed(anActionEvent);
-        });
+
+        Operation operation = TelemetryManager.createOperation(serviceName, operationName);
+        operation.start();
+        EventUtil.logEvent(EventType.info, operation, buildProp(anActionEvent, null));
+        if (onActionPerformed(anActionEvent, operation)) {
```

Comment:
Shall we use a try-finally block here, in case some RuntimeException and we could get the exception, send error telemetry and close the operation.